cometsdk.go

```diff
@@ -3948,11 +3952,21 @@ func (this *CometAPIClient) Request(contentType, method, path string, data map[s
 		if data != nil {
 			body = url.Values(data)
 		}
-		body.Set("AuthType", "Password")
-		body.Set("Username", this.Username)
-		body.Set("Password", this.Password)
+		body.Set("Username", c.Username)
+		body.Set("Password", c.Password)
+		if c.SessionKey != "" {
+			body.Set("AuthType", "SessionKey")
+			body.Set("SessionKey", c.SessionKey)
+		} else if c.TOTPKey != "" {
+			body.Set("AuthType", "PasswordTOTP")
```

Comment:
This is the main code where TOTP and SessionKey based login is introduced

Reply:
Thanks, this is a really nice improvement.